lib/limiter/limiter.go

```diff
@@ -180,7 +180,10 @@ func (l *Limiter) StreamServerInterceptor(srv interface{}, serverStream grpc.Ser
 
 // WrapListener returns a [Listener] that wraps the provided listener
 // with one that limits connections
-func (l *Limiter) WrapListener(ln net.Listener) *Listener {
+func (l *Limiter) WrapListener(ln net.Listener) net.Listener {
```

Comment:
Should we be adding nil checks at the call-site now?

Reply:
Changed `WrapListener` to return an error instead.